reader/unmarshal.go

```diff
@@ -152,24 +164,11 @@ func unmarshalDocument(line []byte) (interface{}, error) {
 }
 
 func unmarshalRange(line []byte) (interface{}, error) {
-	type _position struct {
-		Line      int `json:"line"`
-		Character int `json:"character"`
-	}
-	var payload struct {
-		Start _position `json:"start"`
-		End   _position `json:"end"`
-	}
+	var payload Range
```

Comment:
This may have been previously split for a performance optimization. I'll run this after a round of feedback to see if it still makes a big difference in reading.

Reply:
Ah got it, happy to switch back. Is there a benchmark we could add that tracks this?

Reply:
I can add one.

Reply:
I went ahead and inlined the unmarshalling types in this function to try to save you some trouble. However, I wasn't sure how much of the performance benefit was due to inlining the types vs. not importing anything. For instance, I used an inline field type of [`Tags []protocol.SymbolTag`](https://github.com/sourcegraph/lsif-protocol/pull/6/files#diff-153dea3919d8ce6f346ce5fb3b420bbee3d60b6df58868ce379691ef6a7f63aaR181) instead of `Tags []int`, because the latter would have required creating a new slice and copying over the integer vales one by one. PTAL to see if this satisfies the concern.

I also added a test for unmarshalling a range with a tag (necessary now to test the conversion from inlined type value to exported type value).